### PR TITLE
cmake: cleanup outdated code

### PR DESCRIPTION
**Pull request by apwojcik**

### CODE REVIEW

**Review by junliume — APPROVED** [review]
LGTM but @atamazov could you give it a quick look?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -667,23 +661,13 @@ clang_tidy_check(MIOpen)
...
-    set(PASS_ARGS debug optimized)
-    set(DEPS)
-    foreach(DEP ${ARGN})
-        if(DEP IN_LIST PASS_ARGS)
-            list(APPEND DEPS ${DEP})
-        else()
-            list(APPEND DEPS $<BUILD_INTERFACE:${DEP}>)
-        endif()
-    endforeach()
-    target_link_libraries(${TARGET} INTERFACE ${DEPS})
```
Comment (atamazov) [inline src/CMakeLists.txt:704]: @pfultz2 Do we no longer need this patch for "debug" and "optimized"? I am not quite confident with this stuff.

**Review by atamazov — COMMENTED** [review]
I have no objections, but one mandatory question -- https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2349/files#r1320859644

File: src/CMakeLists.txt (hunk 1 above)
Comment (junliume) [reply src/CMakeLists.txt:704]: Ping @pfultz2 and @JehandadKhan

Comment (junliume) [reply src/CMakeLists.txt:704]: Ping @pfultz2 and @JehandadKhan for review

Comment (pfultz2) [reply src/CMakeLists.txt:704]: No, it isnt used anymore. I think this was used before we could use boost cmake targets.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**